solana-programs/audius_eth_registry/src/processor.rs

```diff
@@ -132,8 +329,267 @@ impl Processor {
+        if instruction_recovery.is_err() {
+            return Err(AudiusError::Secp256InstructionLosing.into());
+        }
+
+        let recovered_instructions = instruction_recovery.unwrap();
+        if recovered_instructions.len() < 3 {
+            return Err(AudiusError::Secp256InstructionLosing.into());
+        }
+
+        let secp_instruction_1 = &recovered_instructions[0];
+        let secp_instruction_2 = &recovered_instructions[1];
+        let secp_instruction_3 = &recovered_instructions[2];
+
+        let signer_group = Box::new(SignerGroup::try_from_slice(
+            &signer_group_info.data.borrow(),
+        )?);
+
+        if !signer_group.is_initialized() {
+            return Err(AudiusError::UninitializedSignerGroup.into());
+        }
+
+        // Create and write new valid signer
+        let mut new_valid_signer = Box::new(ValidSigner::try_from_slice(
+            &new_valid_signer_info.data.borrow(),
+        )?);
+
+        if new_valid_signer.is_initialized() {
+            return Err(AudiusError::SignerAlreadyInitialized.into());
+        }
+
+        let valid_signer_1 = Box::new(ValidSigner::try_from_slice(
+            &valid_signer_1_info.data.borrow(),
+        )?);
+        let valid_signer_2 = Box::new(ValidSigner::try_from_slice(
+            &valid_signer_2_info.data.borrow(),
+        )?);
+        let valid_signer_3 = Box::new(ValidSigner::try_from_slice(
+            &valid_signer_3_info.data.borrow(),
+        )?);
+
+        if !valid_signer_1.is_initialized()
+            || !valid_signer_2.is_initialized()
+            || !valid_signer_3.is_initialized()
+        {
+            return Err(AudiusError::ValidSignerNotInitialized.into());
+        }
+
+        if valid_signer_1.signer_group != *signer_group_info.key
+            || valid_signer_2.signer_group != *signer_group_info.key
+            || valid_signer_3.signer_group != *signer_group_info.key
+        {
+            return Err(AudiusError::WrongSignerGroup.into());
+        }
+
+        let instruction_data_1 = Self::recover_instruction_data(&signature_data_1, &valid_signer_1);
+        let instruction_data_2 = Self::recover_instruction_data(&signature_data_2, &valid_signer_2);
+        let instruction_data_3 = Self::recover_instruction_data(&signature_data_3, &valid_signer_3);
+
+        if instruction_data_1 != secp_instruction_1.data {
+            return Err(AudiusError::SignatureVerificationFailed.into());
+        }
+
+        if instruction_data_2 != secp_instruction_2.data {
+            return Err(AudiusError::SignatureVerificationFailed.into());
+        }
+
+        if instruction_data_3 != secp_instruction_3.data {
+            return Err(AudiusError::SignatureVerificationFailed.into());
+        }
+
+        // Each signature data message is expected to be a recent unix timestamp
+        // If messages do not adhere to this format, the operation will fail
+        let timestamp_result = Self::validate_timestamp_messages(
```

Comment:
I think the whole block of code here is exactly the same as in clear valid signer function and can be refactored to use a common function.